judge/judgedaemon.main.php

```diff
@@ -947,8 +941,13 @@ function compile(array $judgeTask, string $workdir, string $workdirpath, array $
     }
 
     // Compile the program.
-    $compile_cmd = escapeshellcmd(LIBJUDGEDIR . "/compile.sh $cpuset_opt '$execrunpath' '$workdir' " .
-                                  implode(' ', $files));
+    $compile_cmd = LIBJUDGEDIR . "/compile.sh $cpuset_opt " .
+        implode(' ', array_map('dj_escapeshellarg', array_merge([
+            $execrunpath,
+            $workdir,
+        ], $files)));
+    logmsg(LOG_NOTICE, "Compile command: ".$compile_cmd);
+    logmsg(LOG_NOTICE, "Compile files: ".implode(' ', $files));
```

Comment:
this is part of the previous command, isn't it?

Reply:
> this is part of the previous command, isn't it?

Indeed, and that's leftover debug stuff. Changed it into a single `LOG_DEBUG` message.